src/components/KYCWall/BaseKYCWall.js

```diff
@@ -93,10 +95,19 @@ class KYCWall extends React.Component {
      */
     selectPaymentMethod(paymentMethod) {
         this.props.onSelectPaymentMethod(paymentMethod);
-        if (paymentMethod === CONST.PAYMENT_METHODS.BANK_ACCOUNT) {
+        if (paymentMethod === CONST.PAYMENT_METHODS.PERSONAL_BANK_ACCOUNT) {
             Navigation.navigate(this.props.addBankAccountRoute);
         } else if (paymentMethod === CONST.PAYMENT_METHODS.DEBIT_CARD) {
             Navigation.navigate(this.props.addDebitCardRoute);
+        } else if (paymentMethod === CONST.PAYMENT_METHODS.BUSINESS_BANK_ACCOUNT) {
+            if (ReportUtils.isIOUReport(this.props.iouReport)) {
```

Comment:
Is this additional check necessary? We also make the same check within the function. 

I think it is because of the navigation. And it's better to check within the function instead of outside. But I just want to double-check